src/backends/interflop-mca/interflop_mca.c

```diff
@@ -643,9 +591,14 @@ void print_information_header(void *context) {
                 : (ctx->relErr && ctx->absErr)
                       ? MCA_ERR_MODE_STR[mca_err_mode_all]
                       : MCA_ERR_MODE_STR[mca_err_mode_rel],
-      key_rng_mode_str, MCA_RNG_MODE_STR[MCALIB_RNG_MODE], key_err_exp_str,
-      (ctx->absErr_exp), key_daz_str, ctx->daz ? "true" : "false", key_ftz_str,
-      ctx->ftz ? "true" : "false", key_sparsity_str, ctx->sparsity);
+      key_err_exp_str, (ctx->absErr_exp), key_daz_str, ctx->daz ? "true" : "false", 
+      key_ftz_str, ctx->ftz ? "true" : "false", key_sparsity_str, ctx->sparsity);
+}
+
+void _interflop_finalize(__attribute__((unused)) void *context) {
+  /* destroy the global thread id access mutex */
+  if (pthread_mutex_destroy(&global_tid_lock) != 0)
```

Comment:
Using a static pthread initialiser avoids having to call mutex_destroy. So you can simplify this.